dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueInternalWeakReducingGauge.java

```diff
@@ -39,38 +38,44 @@
 @ThreadSafe
 public final class DialogueInternalWeakReducingGauge<T> implements Gauge<Number> {
 
-    @GuardedBy("this")
-    private final Set<T> weakSet = Collections.newSetFromMap(new WeakHashMap<>(2));
-
-    private final ToLongFunction<T> gaugeFunction;
-    private final Function<LongStream, Number> operator;
+    private final Set<T> weakSet = Collections.synchronizedSet(Collections.newSetFromMap(new WeakHashMap<>(2)));
+    private final Function<Stream<T>, Number> summarize;
 
     @VisibleForTesting
-    DialogueInternalWeakReducingGauge(ToLongFunction<T> gaugeFunction, Function<LongStream, Number> reduceFunction) {
-        this.gaugeFunction = gaugeFunction;
-        this.operator = reduceFunction;
+    DialogueInternalWeakReducingGauge(Function<Stream<T>, Number> summarize) {
+        this.summarize = summarize;
     }
 
     /** Register a new source element which will be used to compute the future summary integer. */
-    public synchronized void add(T sourceElement) {
+    public void add(T sourceElement) {
         weakSet.add(sourceElement);
     }
 
     @Override
-    public synchronized Number getValue() {
-        return operator.apply(weakSet.stream().mapToLong(gaugeFunction));
+    public Number getValue() {
+        return summarize.apply(weakSet.stream());
```

Comment:
I don’t think this is valid without method level synchronization, the synchronized wrappers don’t provide any protection for iterators or streams.